src/Hosting/WindowsServices/src/WindowsServiceLifetimeOptions.cs

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Extensions.Hosting;
+
+namespace Microsoft.Extensions.Hosting
+
+{
+    public class WindowsServiceLifetimeOptions : HostOptions
```

Comment:
I don't know how well this derivation will work. You did this to get ShutdownTimeout, right? But now you have two different options objects.

@HaoK How well does `Configure<TOptions>` work with derived types?

You might be better of injecting both options separately.

Reply:
Configure doesn't look at inheritance of types at all (other than what you get for free from the ctor), but only the Configures that are registered against the exact TOptions type are run

Reply:
I was afraid of that. Ok, this model won't work. Split WindowsServiceLifetimeOptions and HostOptions and inject them separately.